src/ctapipe/io/hdf5merger.py

```diff
@@ -268,19 +267,15 @@ def __call__(self, other: str | Path | tables.File):
             if self._n_merged == 0:
                 self.meta = self._read_meta(other)
                 self.data_model_version = self.meta.product.data_model_version
+                self.data_category = self.meta.product.data_category
                 metadata.write_to_hdf5(self.meta.to_dict(), self.h5file)
+                self.required_nodes = self._get_required_nodes(self.h5file)
             else:
                 self._check_can_merge(other)
 
             Provenance().add_input_file(other.filename, "data product to merge")
             try:
                 self._append(other)
-                # if first file, update required nodes
-                if self.required_nodes is None:
```

Comment:
This needs to stay here, after the first `_append`.

Reply:
and the none check _here_ is actually required since it needs to distinguish "unset" from "no nodes required".